<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>623</width>
    <height>427</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="tabletTracking">
   <bool>true</bool>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::CustomContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>601</width>
      <height>351</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <widget class="QToolBox" name="toolBox">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="listPage">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>168</width>
        <height>273</height>
       </rect>
      </property>
      <attribute name="label">
       <string>QListWidget</string>
      </attribute>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>161</width>
         <height>251</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QToolButton" name="tBtnListInit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>初始化列表</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QToolButton" name="tBtnListClear">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>清除列表</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QToolButton" name="tBtnListInsert">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>插入项</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QToolButton" name="tBtnListAppend">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>添加项</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QToolButton" name="tBtnListDelete">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>删除当前项</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="treePage">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>168</width>
        <height>273</height>
       </rect>
      </property>
      <attribute name="label">
       <string>QTreeWidget</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="tablePage">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>168</width>
        <height>273</height>
       </rect>
      </property>
      <attribute name="label">
       <string>QTableWidget</string>
      </attribute>
     </widget>
    </widget>
    <widget class="QTabWidget" name="tabWidget">
     <property name="mouseTracking">
      <bool>true</bool>
     </property>
     <property name="tabletTracking">
      <bool>true</bool>
     </property>
     <property name="contextMenuPolicy">
      <enum>Qt::CustomContextMenu</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabList">
      <attribute name="title">
       <string>QlistWidget</string>
      </attribute>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>11</y>
         <width>371</width>
         <height>301</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>当前项变化</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="editCurrentItem"/>
          </item>
          <item>
           <widget class="QCheckBox" name="chkListEditable">
            <property name="text">
             <string>可编辑</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QToolButton" name="tBtnSelItem">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>项选择</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tBtnSelAll">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>全选</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tBtnSelNone">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>全不选</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tBtnSelInverse">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>反选</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="listWidget">
            <property name="mouseTracking">
             <bool>false</bool>
            </property>
            <property name="tabletTracking">
             <bool>false</bool>
            </property>
            <property name="contextMenuPolicy">
             <enum>Qt::CustomContextMenu</enum>
            </property>
            <item>
             <property name="text">
              <string>New Item</string>
             </property>
             <property name="checkState">
              <enum>Unchecked</enum>
             </property>
             <property name="flags">
              <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
             </property>
            </item>
            <item>
             <property name="text">
              <string>New Item2</string>
             </property>
             <property name="checkState">
              <enum>Unchecked</enum>
             </property>
             <property name="flags">
              <set>ItemIsSelectable|ItemIsEditable|ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tabTree">
      <attribute name="title">
       <string>QTreeWidget</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="tabTable">
      <attribute name="title">
       <string>QTableWidget</string>
      </attribute>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>623</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actListInit"/>
   <addaction name="separator"/>
   <addaction name="actListClear"/>
   <addaction name="separator"/>
   <addaction name="actListInsert"/>
   <addaction name="separator"/>
   <addaction name="actListAppend"/>
   <addaction name="separator"/>
   <addaction name="actListDel"/>
   <addaction name="separator"/>
  </widget>
  <action name="actListInit">
   <property name="text">
    <string>初始化列表</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actListClear">
   <property name="text">
    <string>清除列表</string>
   </property>
   <property name="toolTip">
    <string>清除列表</string>
   </property>
  </action>
  <action name="actListInsert">
   <property name="text">
    <string>插入项</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actListAppend">
   <property name="text">
    <string>添加项</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actListDel">
   <property name="text">
    <string>删除当前项</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actSelAll">
   <property name="text">
    <string>全选</string>
   </property>
  </action>
  <action name="actSelNone">
   <property name="text">
    <string>全不选</string>
   </property>
  </action>
  <action name="actSelInvs">
   <property name="text">
    <string>反选</string>
   </property>
  </action>
  <action name="actQuit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="toolTip">
    <string>退出程序</string>
   </property>
  </action>
  <action name="actSelPopMenu">
   <property name="text">
    <string>项选择</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actSelPopMenu</sender>
   <signal>triggered()</signal>
   <receiver>actSelInvs</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnListInit</sender>
   <signal>clicked()</signal>
   <receiver>actListInit</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnListClear</sender>
   <signal>clicked()</signal>
   <receiver>actListClear</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnListInsert</sender>
   <signal>clicked()</signal>
   <receiver>actListInsert</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnListAppend</sender>
   <signal>clicked()</signal>
   <receiver>actListAppend</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnListDelete</sender>
   <signal>clicked()</signal>
   <receiver>actListDel</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnSelAll</sender>
   <signal>clicked()</signal>
   <receiver>actSelAll</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actListClear</sender>
   <signal>triggered()</signal>
   <receiver>listWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnSelNone</sender>
   <signal>clicked()</signal>
   <receiver>actSelNone</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tBtnSelInverse</sender>
   <signal>clicked()</signal>
   <receiver>actSelInvs</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>544</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
